Documentation/CRDs/Cluster/external-cluster/topology-for-external-mode.md

```diff
@@ -0,0 +1,199 @@
+# Topology provisioning for external mode
+
+## Background
+For Block storage (RWO), Enable Topology awareness for PV provisioning. Preferably, picking the replica 1 Storage Class for this AZ or creating a volume in a relevant AZ using a single storage class. It will help in providing Isolation to application/pods in AZ
+
+## Scenario
+Applications like Kafka will have a deployment with multiple running instances. Each service instance will create a new claim and is expected to be located in a different zone. Since the application has its own redundant instances, there is no requirement for redundancy at the data layer. A storage class is created that will provision storage from replica 1 Ceph pools that are located in each of the separate zones.
+
+## Technical implementation details
+
+Add --topology-pools, --topology-failure-domain-label, --topology-failure-domain-values to the script: `create-external-cluster-resources.py` where,
+`--topology-pools`: (optional) comma-separated list of topology-constrained rbd pools
+`--topology-failure-domain-label`: (optional) k8s cluster failure domain label (example: zone,rack,host,etc) for the topology-pools that are matching the ceph domain
+ `--topology-failure-domain-values`: (optional) comma-separated list of the k8s cluster failure domain values corresponding to each of the pools in the topology-pools list
+
+The import script will then create a new storage class named `ceph-rbd-topology`.
+
+## Example Configuration
+
+### Ceph cluster
+
+Looking at the osd tree, before jumping to the steps
+```
```

Comment:
No need to include a full CRUSH map. Ceph users will understand without seeing a full example.